src/Bundle/Maker/MakeFactoryData.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Zenstruck\Foundry\Bundle\Maker;
+
+use Zenstruck\Foundry\ModelFactory;
+use Zenstruck\Foundry\Proxy;
+
+final class MakeFactoryData
+{
+    /** @var list<string> */
+    private array $uses;
+    /** @var array<string, string> */
+    private array $defaultProperties = [];
+    /** @var list<MakeFactoryPHPDocMethod> */
+    private array $methodsInPHPDoc;
+
+    public function __construct(private \ReflectionClass $object, private ?\ReflectionClass $repository, private bool $withPHPStanEnabled, private bool $persisted)
+    {
+        $this->uses = [
+            ModelFactory::class,
+            Proxy::class,
+            $object->getName(),
+        ];
+
+        if ($repository) {
+            $this->uses[] = $repository->getName();
+        }
+
+        $this->methodsInPHPDoc = $this->createMethodsForPHPDoc($persisted);
+    }
+
+    public function getObject(): \ReflectionClass
+    {
+        return $this->object;
+    }
+
+    public function getObjectShortName(): string
+    {
+        return $this->object->getShortName();
+    }
+
+    /** @return class-string */
+    public function getObjectFullyQualifiedClassName(): string
+    {
+        return $this->object->getName();
+    }
+
+    public function getRepositoryShortName(): ?string
+    {
+        return $this->repository?->getShortName();
+    }
+
+    public function isPersisted(): bool
+    {
+        return $this->persisted;
+    }
+
+    public function addUse(string $use): void
+    {
+        if (!\in_array($use, $this->uses, true)) {
+            $this->uses[] = $use;
+        }
+    }
+
+    public function renderUses(): string
```

Comment:
don't know if the clearer is to return a string or an array here (same for other `render*()` methods

Reply:
Yeah, this is getting pretty _templatey_. Thinking the rendering should be done in the tpl file.